modules/KalturaSupport/resources/mw.QualityWatcher.js

```diff
@@ -0,0 +1,106 @@
+            var _this = this;
+            var vidObj = _this.embedPlayer.getVideoHolder()[0].getElementsByTagName("video")[0];
+            if (typeof vidObj.getVideoPlaybackQuality === 'function') {
+                var videoPlaybackQuality = vidObj.getVideoPlaybackQuality();
+                _this.checkFPS( videoPlaybackQuality.droppedVideoFrames , videoPlaybackQuality.totalVideoFrames );
+            } else {
+                _this.checkFPS( vidObj.webkitDroppedFrameCount , vidObj.webkitDecodedFrameCount );
+            }
+        },
+        checkFPS: function ( droppedFrames, decodedFrames) {
+            try {
+                var currentTime = performance.now();
+                if (decodedFrames) {
+                    if (this.lastTime) {
+                        var currentPeriod = currentTime - this.lastTime,
+                            currentDropped = droppedFrames - this.lastDroppedFrames,
+                            currentDecoded = decodedFrames - this.lastDecodedFrames,
+                            droppedFPS = 1000 * currentDropped / currentPeriod;
+                        if (droppedFPS > 0) {
+                            this.log('checkFPS : droppedFPS/decodedFPS:' + droppedFPS / (1000 * currentDecoded / currentPeriod));
+                            if (currentDropped > (mw.getConfig("fpsDroppedMonitoringThreshold") || 0.1) * currentDecoded) {
+                                var currentLevel = this.getCurrentQuality();
+                                this.log('drop FPS ratio greater than max allowed value for currentLevel: ' + currentLevel);
+                                if (currentLevel > 0) {
+                                    this.removeQuality(currentLevel);
+                                }
+                            }
+                        }
+                    }
+                    this.lastTime = currentTime;
+                    this.lastDroppedFrames = droppedFrames;
+                    this.lastDecodedFrames = decodedFrames;
+                }
+            } catch(e){this.log("Error occur while trying to check dropFrames");}
+        },
+        getSortedTracks: function(){
+            var tracks = player.getVariantTracks();
+            var sortedTracks = tracks.map((obj)=>{return {id:obj.id,bandwidth:obj.bandwidth,active:obj.active}}).sort((obj1,obj2)=>{return obj1.bandwidth>obj2.bandwidth});
+            return sortedTracks;
+        },
+        getCurrentQuality:function(){
+            var sortedTracks = this.getSortedTracks();
+            var activeTrackId = 0;
+            for (var index=0 ; index < sortedTracks.length ; index++){
+                if (sortedTracks[index].active){
+                    activeTrackId = index;
+                }
+            }
+            return activeTrackId;
+        },
+        removeQuality:function(level){
+            var sortedTracks = this.getSortedTracks();
+            var bandWidthToRemove = sortedTracks[level].bandwidth;
```

Comment:
I think you can get the current variant bandwidth can be achived also by calling `getStats` API.
The value is under `streamBandwidth`, so you might be able to avoid all the logic to look for current video track.